### PR TITLE
fix(Spin): Fixed the indicator itself percent attribute being invalid

### PR DESCRIPTION
**Pull request by wanpan11**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- ref https://github.com/ant-design/ant-design/pull/49211
- demo https://stackblitz.com/edit/react-bmmhq9-jsygeb?file=demo.tsx

### 💡 Background and solution

使用 Progress 作为指示器时 原生的 percent 无效

image

### 📝 Changelog

- Use a developer-oriented tone and narrative style.
- Describe the user's first-hand experience of the issue and its impact on developers, rather than your solution approach.
- Refer to: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fixed the indicator itself percent attribute being invalid      |
| 🇨🇳 Chinese |      修复自定义指示器 percent 属性无效问题     |

### CODE REVIEW

File: components/spin/Indicator/index.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ export default function Indicator(props: IndicatorProps) {
   if (indicator && React.isValidElement(indicator)) {
     return cloneElement(indicator, {
       className: classNames(indicator.props.className, dotClassName),
-      percent,
+      percent: indicator.props.percent || percent,
```
Comment (zombieJ) [inline components/spin/Indicator/index.tsx:20]: 做个 feature 允许 indicator 传入一个 renderProps 吧~~

**Comment by afc163** [comment]
是 https://github.com/ant-design/ant-design/issues/50162 这个问题么？

**Comment by wanpan11** [comment]
> 是 #50162 这个问题么？

不确定是不是相同的问题，是想修上次 PR 造成的 breaker change 来着 🤧

**Comment by afc163** [comment]
看看 https://github.com/ant-design/ant-design/issues/50162#issuecomment-2261792391 这个结论，也有一定的道理。

**Comment by wanpan11** [comment]
我先把这个关掉吧

**Pull request closed without merging**